core/src/main/scripts/importer/importOncokbDiscreteCNA.py

```diff
@@ -0,0 +1,288 @@
+                             "OncoKb annotations will not be imported for this gene." \
+                             "Please fix and rerun.",
+                             extra={'symbol': feature['Hugo_Symbol']})
+                feature['Entrez_Gene_Id'] = None
+            elif len(entrez_gene_ids) == 0:
+                logger.error("Could not find the Entrez gene id for a gene." \
+                             "OncoKb annotations will not be imported for this gene." \
+                             "Please fix and rerun.",
+                             extra={'symbol': feature['Hugo_Symbol']})
+                feature['Entrez_Gene_Id'] = None
+            else:
+                feature['Entrez_Gene_Id'] = str(entrez_gene_ids[0])
+                feature['id'] = "_".join([feature['Entrez_Gene_Id'], feature['copyNameAlterationType']])
+
+            features.append(feature)
+    cna_file.close()
+    print(" DONE")
+    return features
+
+
+def fetch_oncokb_annotations(features):
+    """Submit CNA events to OncoKB.org and return OncoKB annotations."""
+    id_to_annotation = {}
+    payload_list = create_request_payload(features)
+    annotations = libImportOncokb.fetch_oncokb_annotations(payload_list, "https://public.api.oncokb.org/api/v1/annotate/copyNumberAlterations")
+    for annotation in annotations:
+        id = annotation['query']['id']
+        id_to_annotation[id] = annotation
+    return id_to_annotation
+
+
+def create_request_payload(features):
+    """Translate CNA events into JSON for message body."""
+    elements = {}
+    for feature in features:
+        elements[feature[
+            'id']] = '{ "copyNameAlterationType":"%s", "gene":{"entrezGeneId":%s}, "id":"%s", "tumorType":null} ' \
+                     % (feature['copyNameAlterationType'], feature['Entrez_Gene_Id'], feature['id'])
+    # normalize for alteration id since same alteration is represented in multiple samples
+    return list(elements.values())
+
+
+def update_cna_metafile(meta_cna_file_path, pd_file_name):
+    """Add reference to pd annotation file to CNA meta file."""
+    meta_file = open(meta_cna_file_path, "r")
+    lines = meta_file.readlines()
+    meta_file.close()
+    meta_file_name = os.path.basename(meta_cna_file_path)
+    dir = os.path.dirname(meta_cna_file_path)
+    os.rename(meta_cna_file_path, os.path.join(dir, 'ONCOKB_IMPORT_BACKUP_' + meta_file_name))
+    if lines[-1] == '\n':
+        lines = lines[:-1]
+    meta_file = open(meta_cna_file_path, "w")
+    for line in lines:
+        meta_file.write(line)
+    meta_file.write('pd_annotations_filename: ' + pd_file_name + '\n')
+    meta_file.close()
+
+
+def write_annotations_to_file(features, pd_file_name):
+    """Write CNA pd annotations to data file."""
+    new_file = open(pd_file_name, "w")
+    new_file.write("SAMPLE_ID\tEntrez_Gene_Id\tcbp_driver\tcbp_driver_annotation\tcbp_driver_tiers\tcbp_driver_tiers_annotation\n")
```

Comment:
It seems strange to me you have all upper case, camel case, and lower case headers. Is this your intention?

Reply:
This is according to specification in FileFormats.md. I think that this discussion is out of scope for this PR.

Reply:
Ah nice! I don't even know that file exists. Thanks for the info!